go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/microsoft/retina
 
-go 1.22.4
+go 1.23.1
```

Comment:
Unless there's something materially different about the language in 1.23 that Retina depends on, we should keep this at the minimum version necessary to build Retina (as defined in the spec).

To version bump, we should set `toolchain 1.23.1`
> toolchain
> 
> Declares a suggested Go toolchain to use with this module. Only takes effect when the module is the main module and the default toolchain is older than the suggested toolchain.